src/auth/auth.service.ts

```diff
@@ -33,4 +38,28 @@ export class AuthService {
 
     return this.userService.create(userDto);
   }
+
+  async login(user: JwtUser) {
+    const token = this.jwtService.sign(user);
+
+    return {
+      accessToken: token,
+      statusCode: HttpStatus.CREATED,
+    };
+  }
+
+  async validateUser(username: string, pass: string) {
+    const user: UserEntity | null = await this.userService.findOne(username);
+
+    if (user !== null) {
+      const passwordCorrect = bcrypt.compareSync(pass, user.password);
+      const userData = { email: user.email, id: user.id };
+
+      if (passwordCorrect) {
+        return userData;
+      }
+    }
```

Comment:
should better follows the return early pattern, it could improve readability of code.